arangod/Cluster/ClusterMethods.cpp

```diff
@@ -2881,109 +2881,148 @@ std::vector<std::shared_ptr<LogicalCollection>> ClusterMethods::persistCollectio
     std::vector<std::shared_ptr<LogicalCollection>>& collections,
     bool ignoreDistributeShardsLikeErrors, bool waitForSyncReplication,
     bool enforceReplicationFactor) {
+  
   TRI_ASSERT(!collections.empty());
   if (collections.empty()) {
     THROW_ARANGO_EXCEPTION_MESSAGE(
         TRI_ERROR_INTERNAL,
         "Trying to create an empty list of collections on coordinator.");
   }
+  
+  double const realTimeout = ClusterInfo::getTimeout(240.0);
+  double const endTime = TRI_microtime() + realTimeout;
+
   // We have at least one, take this collections DB name
-  auto& dbName = collections[0]->vocbase().name();
+  auto const dbName = collections[0]->vocbase().name();
```

Comment:
I think it should be clear, because all operations a user can carry out (e.g. creating a collection, or multiple collections at the same time) are always executed in the context of a single database.
We can put an extra comment here if you think it makes sense.

Reply:
Added the comment